[dev_tools] Flag when a package has no top in `check_ir_equivalence_main`.

cdleary opened the pull request:
Previously this would segfault.

Also adds pylint to the pre-commit yaml; this effectively fixes #1986 as we will now see pylint errors via the pre-commit hook

allight commented on xls/dev_tools/check_ir_equivalence_main_test.py:
```diff
@@ -135,34 +143,39 @@ def _check_equiv(
 
   def test_detects_different_procs(self):
-    res, msg = self._check_equiv(PROC_IR, NEG_PROC_IR, act_count=14)
+    res, stdout, stderr = self._check_equiv(PROC_IR, NEG_PROC_IR, act_count=14)
```
the `stderr` should be `_stderr` in all of these so that the unused variable lint doesn't trigger.

grebe commented on xls/dev_tools/check_ir_equivalence_main_test.py:
```diff
@@ -135,34 +143,39 @@ def _check_equiv(
+    self.assertIn("Verified equivalent", stdout)
 
+  def test_lhs_has_no_top(self):
+    res, msg, stderr = self._check_equiv(ADD_IR, ADD_NO_TOP_IR)
```
Copybara doesn't like all the unused variables (mostly stderr, this last msg is a problem). We can fix it up internally, but you might be able to do it faster. LMK

grebe replied:
Ah, nevermind lol. Didn't see Alex's comment

cdleary commented:
Okay I added pylintrc to the repo and a pre-commit hook for it per the discussion in #1986 cc @meheff and @mikex-oss

cdleary commented:
Oh it was already in the repo! Reverted.

meheff commented:
Thanks for adding the lint check! I assume next step is for us to expand the regex to cover all python files and fix any issues?

> Oh it was already in the repo! Reverted.

What part got reverted?

mikex-oss commented:
> Thanks for adding the lint check! I assume next step is for us to expand the regex to cover all python files and fix any issues?
> 
> > Oh it was already in the repo! Reverted.
> 
> What part got reverted?

Looks like pylintrc is already in our repo. Might be stale though.

cdleary commented:
> > Thanks for adding the lint check! I assume next step is for us to expand the regex to cover all python files and fix any issues?
> > > Oh it was already in the repo! Reverted.
> > 
> > 
> > What part got reverted?
> 
> Looks like pylintrc is already in our repo. Might be stale though.

It wasn't stale, there was no meaningful change when I pulled the most recent one aside from switching out our tabs-as-spaces configuration (we use 2 whereas canonical now is 4). So I just reverted to the one that was in there, seems legit.

cdleary commented:
This is rebased now, PTAL.
